app/src/main/java/com/celzero/bravedns/service/ConnectionMonitor.kt

```diff
@@ -0,0 +1,242 @@
+                    connectionMonitor.currentActiveNetwork = activeNetwork
+                }
+            } else {
+                connectionMonitor.networkListener.onNetworkDisconnected()
+                connectionMonitor.currentActiveNetwork = null
+            }
+            if (DEBUG) Log.d(LOG_TAG, "ConnectionMonitor - ${connectionMonitor.connectivityManager.getNetworkInfo(activeNetwork)?.typeName.toString()}")
+            if (DEBUG) Log.d(LOG_TAG, "ConnectionMonitor - ${connectionMonitor.connectivityManager.getNetworkInfo(connectionMonitor.currentActiveNetwork)?.typeName.toString()}")
+        }
+
+        /**
+         * Adds all the available network to the underlying network.
+         */
+        private fun processAllAvailableNetwork() {
+            var isActiveNetworkChanged = false
+            setAvailableNetworks()
+            if (DEBUG) Log.d(LOG_TAG, "ConnectionMonitor - process message MESSAGE_AVAILABLE_NETWORK")
+            if (connectionMonitor.currentNetworkList.size > 0 && addedNetworkList.size > 0) {
+                if (DEBUG) Log.d(LOG_TAG, "ConnectionMonitor - ${connectionMonitor.connectivityManager.getNetworkInfo(connectionMonitor.currentNetworkList.elementAt(0))?.typeName.toString()}")
+                if (DEBUG) Log.d(LOG_TAG, "ConnectionMonitor - ${connectionMonitor.connectivityManager.getNetworkInfo(addedNetworkList.elementAt(0))?.typeName.toString()}")
+                isActiveNetworkChanged = (connectionMonitor.currentNetworkList.elementAt(0).networkHandle != addedNetworkList.elementAt(0).networkHandle)
+            }
+            val setDiff = Sets.symmetricDifference(connectionMonitor.currentNetworkList, addedNetworkList)
+            if (setDiff.size != 0 || isActiveNetworkChanged) {
+                connectionMonitor.networkListener.onNetworkChange(addedNetworkList)
+                connectionMonitor.currentNetworkList.clear()
+                connectionMonitor.currentNetworkList.addAll(addedNetworkList)
+            }
+        }
+
+        /**
+         * The first element of the list will be the active network.
+         */
+        private fun setAvailableNetworks() {
+            addedNetworkList = linkedSetOf()
+            connectionMonitor.connectivityManager.activeNetwork?.let {
+                addedNetworkList.add(it)
+            }
+            val tempAllNetwork = connectionMonitor.connectivityManager.allNetworks
+            tempAllNetwork.forEach {
+                val hasInternet = connectionMonitor.connectivityManager.getNetworkCapabilities(it)?.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
+                val isVPN = connectionMonitor.connectivityManager.getNetworkCapabilities(it)?.hasTransport(NetworkCapabilities.TRANSPORT_VPN)
+                if (hasInternet == true && isVPN == false) {
+                    addedNetworkList.add(it)
+                }
+            }
+        }
+
+    }
+
```

Comment:
see above

Reply:
Changes for the setAvailableNetworks are completed.